oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/delegate/SessionDelegate.java

```diff
@@ -212,6 +213,8 @@ public void refreshAtNextAccess() {
             } finally {
                 postPerform(sessionOperation, t0);
             }
+        } catch (ReadyOnlyBuilderException e) {
+            throw new RepositoryException(e);
```

Comment:
What is the rationale behind throwing a RepositoryException vs ConstraintViolationException?

I'm not saying it's wrong. Just want to understand the reason why one is used instead of the other.

Reply:
We actually discussed it in the OAK-9612 to throw a ```ConstraintViolationException```. Will change it.